src/app/reporting/Engine.cpp

```diff
@@ -52,9 +59,81 @@ Status EventPathValid(DataModel::Provider * model, const ConcreteEventPath & eve
+///   read should proceed.
+std::optional<CHIP_ERROR> ValidateReadAttributeACL(DataModel::Provider * dataModel, const SubjectDescriptor & subjectDescriptor,
+                                                   const ConcreteReadAttributePath & path)
+{
+
+    RequestPath requestPath{ .cluster     = path.mClusterId,
+                             .endpoint    = path.mEndpointId,
+                             .requestType = RequestType::kAttributeReadRequest,
+                             .entityId    = path.mAttributeId };
+
+    std::optional<DataModel::AttributeInfo> info = dataModel->GetAttributeInfo(path);
+
+    // If the attribute exists, we know whether it is readable (readPrivilege has value)
+    // and what the required access privilege is. However for attributes missing from the metatada
+    // (e.g. global attributes) or completely missing attributes we do not actually know of a required
+    // privilege and default to kView (this is correct for global attributes and a reasonable check
+    // for others)
+    Privilege requiredPrivilege = Privilege::kView;
+    if (info.has_value() && info->readPrivilege.has_value())
+    {
+        // attribute exists and is readable, set the correct read privilege
+        requiredPrivilege = *info->readPrivilege;
+    }
+
+    CHIP_ERROR err = GetAccessControl().Check(subjectDescriptor, requestPath, requiredPrivilege);
+    if (err == CHIP_NO_ERROR)
+    {
+        if (IsSupportedGlobalAttributeNotInMetadata(path.mAttributeId))
+        {
+            // Global attributes passing a kView check is ok
+            return std::nullopt;
+        }
+
+        // We want to return "success" (i.e. nulopt) IF AND ONLY IF the attribute exists and is readable (has read privilege).
+        // Since the Access control check above may have passed with kView, we do another check here:
+        //    - Attribute exists (info has value)
+        //    - Attribute is readable (readProvilege has value) and not "write only"
+        // If the attribute exists and is not readable, we will return UnsupportedRead (spec 8.4.3.2: "Else if the path indicates
+        // attribute data that is not readable, an AttributeStatusIB SHALL be generated with the UNSUPPORTED_READ Status Code.")
+        //
+        // TODO:: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/9024 requires interleaved ordering that
+        //        is NOT implemented here. Spec requires:
+        //           - check cluster access check (done here as kView at least)
+        //           - unsupported endpoint/cluster/attribute check (NOT done here) when the attribute is missing.
+        //             this SHOULD be done here when info does not have a value. This was not done as a first pass to
+        //             minimize amount of delta in the initial PR.
+        //           - "write-only" attributes should return UNSUPPORTED_READ (this is done here)
```

Comment:
Why do we keep doing this wrong against what we need to get to? We should at the same time as we do this update the code to always return UNSUPPORTED_ACCESS first if the access doesn't exist. We know it has to be done, so why do we keep refactoring but doing a different behavior?

Reply:
I did not do this on purpose to limit the scope of this PR. Will make this change once this merges, since I expect it will have sideffects in needing unit and integration test updates.

Reply:
Resolving based on offline chat - this will be a followup (hoping small)